src/fparser/common/readfortran.py

```diff
@@ -210,6 +210,48 @@ def _is_fix_comment(line, isstrict):
 _IS_CALL_STMT = re.compile(r'call\b', re.I).match
 
 
+def extract_label(line):
+    '''Look for an integer label at the start of 'line' and if there is
+    one then remove it from 'line' and return it in 'label'.
+
+    :param str line: a string that potentially contains a label at the \
+        start.
+
+    :returns: a 2-tuple containing the label and updated line if a \
+        label is found or None and the unchanged line if a label is \
+        not found.
+    :rtype: (int or NoneType, str)
+
+    '''
+    label = None
+    match = _LABEL_RE.match(line)
+    if match:
+        label = int(match.group('label'))
+        line = line[match.end():].lstrip()
+    return label, line
+
+
+def extract_construct_name(line):
+    '''Look for a construct name at the start of 'line' and if there is
+    one then remove it from 'line' and return it in 'name'.
```

Comment:
Strictly speaking this routine doesn't return 'name' - it returns an unamed tuple. Please straighten this out (or you could use a named tuple but that's probably more trouble than it's worth).

Reply:
Definitely picky :-). I was describing it at a high level. You can look at the markup to see what it returns. Anyway, I've hopefully updated adequately.